### PR TITLE
feat: Add equality and serialization to poly_triple

### PR DESCRIPTION
**Pull request by phated**
# Description

Noir needs to be able to serialize and deserialize the poly_triple struct. It also needs to be able to check equality of these.

# Checklist:

- [x] I have reviewed my diff in github, line by line.
- [x] Every change is related to the PR description.
- [ ] I have [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) this pull request to the issue(s) that it resolves.
- [x] There are no unexpected formatting changes, superfluous debug logs, or commented-out code.
- [x] There are no circuit changes, OR specifications in `/markdown/specs` have been updated.
- [x] There are no circuit changes, OR a cryptographer has been assigned for review.
- [x] I've updated any terraform that needs updating (e.g. environment variables) for deployment.
- [x] The branch has been rebased against the head of its merge target.
- [x] I'm happy for the PR to be merged at the reviewer's next convenience.
- [ ] New functions, classes, etc. have been documented according to the doxygen comment format. Classes and structs must have `@brief` describing the intended functionality.
- [x] If existing code has been modified, such documentation has been added or updated.

### CODE REVIEW

File: cpp/src/aztec/proof_system/composer/composer_base.hpp [hunk 1]
```diff
@@ -56,8 +57,49 @@ struct poly_triple {
...
+            lhs.a == rhs.a &&
+            lhs.b == rhs.b &&
+            lhs.c == rhs.c &&
+            lhs.q_m == rhs.q_m &&
+            lhs.q_l == rhs.q_l &&
+            lhs.q_r == rhs.q_r &&
+            lhs.q_o == rhs.q_o &&
+            lhs.q_c == rhs.q_c;
+        // clang-format on
+    }
```
Comment (adr1anh) [inline cpp/src/aztec/proof_system/composer/composer_base.hpp]: Is it possible to replace this with `bool operator==(poly_triple const& lhs, poly_triple const& rhs) = default;`?

Comment (kevaundray) [reply cpp/src/aztec/proof_system/composer/composer_base.hpp]: If the minimum version of CPP supported is C++20, then yep we can use default comparisons here instead

Comment (phated) [reply cpp/src/aztec/proof_system/composer/composer_base.hpp]: Checking the CMake file, it looks like C++20 is required, so I can make that change.

**Review by adr1anh — APPROVED** [review]
LGTM

**Pull request merged**